tests/integration/security/sds_egress/sds_istio_mutual_egress_test.go

```diff
@@ -0,0 +1,220 @@
+//  Copyright 2019 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package sdsegress
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/echo/common/response"
+	epb "istio.io/istio/pkg/test/echo/proto"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/components/prometheus"
+	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/pkg/test/util/file"
+	"istio.io/istio/tests/integration/security/util"
+)
+
+const (
+	// should be templated into YAML, and made interchangeable with other sites
+	externalURL      = "http://edition.cnn.com"
+	externalReqCount = 10
+	egressName       = "istio-egressgateway"
+)
+
+var gatewayConfig = `
```

Comment:
Would be nicer to have all CRD coming from same source, inlined string literal or testdata/ folder.

Reply:
different file with duplicated prefix is okay.

Reply:
done